lib/dataset-tools/augmentation/replace_parameters.ts

```diff
@@ -678,7 +671,47 @@ export default class ParameterReplacer {
             }
         }
 
-        for (const slot of program.iterateSlots2()) {
+        const entities : ThingTalk.NNSyntax.EntityMap = {};
+        // replace all entities with SLOT_*, which allows us to pass a VarRef instead of a real value
+        const replaced_code = this._replaceWithSlot(code, entities);
+        const target_program = ThingTalk.NNSyntax.fromNN(replaced_code, entities);
+        await target_program.typecheck(this._schemas, true);
+        
+        if (context != null && context.length > 0) {
```

Comment:
Don't use `!=`, always use `!==`. Also, the context will not be null, undefined, or the empty string. It will be the empty array (see type definition), so you need use `context.length !== 0`.

Also, please run `npx make` and `npm run lint` to find all these problems.
Normally the PR would be blocked because Travis would complain, but Travis is down at the moment...